pkg/schedule/operator/operator_controller.go

```diff
@@ -416,26 +436,28 @@ func (oc *Controller) checkAddOperator(isPromoting bool, ops ...*Operator) bool
 				panic(op)
 			})
 			operatorWaitCounter.WithLabelValues(op.Desc(), "unexpected-status").Inc()
-			return false
+			return false, NotInCreateStatus
 		}
 		if !isPromoting && oc.wopStatus.ops[op.Desc()] >= oc.config.GetSchedulerMaxWaitingOperator() {
 			log.Debug("exceed max return false", zap.Uint64("waiting", oc.wopStatus.ops[op.Desc()]), zap.String("desc", op.Desc()), zap.Uint64("max", oc.config.GetSchedulerMaxWaitingOperator()))
 			operatorWaitCounter.WithLabelValues(op.Desc(), "exceed-max").Inc()
-			return false
+			return false, ExceedWaitLimit
 		}
 
 		if op.SchedulerKind() == OpAdmin || op.IsLeaveJointStateOperator() {
 			continue
 		}
 	}
 	expired := false
+	var reason CancelReasonType
 	for _, op := range ops {
 		if op.CheckExpired() {
 			expired = true
+			reason = Expired
 			operatorWaitCounter.WithLabelValues(op.Desc(), "expired").Inc()
 		}
 	}
```

Comment:
How about delete a var 🤔
```suggestion
	for _, op := range ops {
		if op.CheckExpired() {
			reason = Expired
			operatorWaitCounter.WithLabelValues(op.Desc(), "expired").Inc()
		}
	}
	return reason != Expired, reason
```

Reply:
fixed